authentik/sources/ldap/models.py

```diff
@@ -183,9 +188,47 @@ def connection(
             if server_kwargs.get("get_info", ALL) == NONE:
                 raise exc
             server_kwargs["get_info"] = NONE
-            return self.connection(server_kwargs, connection_kwargs)
+            return self.connection(server, server_kwargs, connection_kwargs)
         return RuntimeError("Failed to bind")
 
+    @property
+    def sync_lock(self) -> Lock:
+        return Lock(cache.client.get_client(), name=f"goauthentik.io/sources/ldap/sync-{self.slug}")
```

Comment:
Does this ever get expired if the app gets killed while having the lock?

Reply:
I thought there was a default inbuilt mechanic that does a keepalive, but seems that is not the case; fixed now